tensorboard/webapp/widgets/line_chart_v2/lib/series_line_view.ts

```diff
@@ -15,19 +15,140 @@ limitations under the License.
 
 import {DataDrawable} from './drawable';
 
+enum PartitionType {
+  NUMBER,
+  NAN,
+}
+
 export class SeriesLineView extends DataDrawable {
+  private partitionPolyline(
+    polyline: Float32Array
+  ): Array<{polyline: Float32Array; type: PartitionType}> {
+    if (polyline.length % 2 !== 0) {
+      throw new Error(`Cannot have odd length-ed polyline: ${polyline.length}`);
+    }
+
+    const partition = [];
+    let partitionStartInd: number = 0;
+    let isPrevValueNaN = false;
+    const zeroCoord = this.coordinator.transformDataToUiCoord(
+      this.getLayoutRect(),
+      [0, 0]
+    );
+    const zeroPoint = {x: zeroCoord[0], y: zeroCoord[1]};
+
+    let lastLegalNumber: {
+      x: number;
+      y: number;
+    } | null = null;
+
+    function recordPartition(
```

Comment:
nit: I think the reading flow might be slightly improved if this function definition was not in the middle of the function body.  Could we move it to either the end or the beginning, or even outside?